tests/lib/util/source-code-fixer.js

```diff
@@ -242,6 +270,221 @@ describe("SourceCodeFixer", function() {
 
         });
 
+        describe("BOM manipulations", function() {
+
+            it("should insert BOM with an insertion of '\uFEFF' at 0", function() {
+                var result = SourceCodeFixer.applyFixes(sourceCode, [ INSERT_BOM ]);
+                assert.equal(result.output, "\uFEFF" + TEST_CODE);
+                assert.isTrue(result.fixed);
+                assert.equal(result.messages.length, 0);
+            });
+
+            it("should insert BOM with an insertion of '\uFEFFfoobar' at 0", function() {
```

Comment:
Fair enough, but having foobar in the quotes makes it look like a string
literal so I assumed that was what you were aiming for. Maybe it could say
something like "... '\uxFEFF' plus any string..." so that only the BOM is
in the string literal? Then again maybe I'm overthinking it.
On Jan 18, 2016 12:51 PM, "Toru Nagashima" <notifications@github.com> wrote:

> In tests/lib/util/source-code-fixer.js
> <https://github.com/eslint/eslint/pull/4935#discussion_r50032422>:
>
> > @@ -242,6 +270,221 @@ describe("SourceCodeFixer", function() {
> >
> >          });
> >
> > +        describe("BOM manipulations", function() {
> > +
> > +            it("should insert BOM with an insertion of '\uFEFF' at 0", function() {
> > +                var result = SourceCodeFixer.applyFixes(sourceCode, [ INSERT_BOM ]);
> > +                assert.equal(result.output, "\uFEFF" + TEST_CODE);
> > +                assert.isTrue(result.fixed);
> > +                assert.equal(result.messages.length, 0);
> > +            });
> > +
> > +            it("should insert BOM with an insertion of '\uFEFFfoobar' at 0", function() {
>
> My intention here, foobar is "any string".
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eslint/eslint/pull/4935/files#r50032422>.
>